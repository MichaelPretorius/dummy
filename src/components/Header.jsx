import React from 'react';
import { Link, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';

import { selectIsAuthenticated } from '../redux/auth/authSelectors';
import { NormalButton } from './CustomButton';
import { logoutUser } from '../redux/auth/authActions';

const Header = ({ isAuthenticated, logoutUser, history }) => {
  return (
    <>
      <Link to="/">Home</Link>
      <Link to="/private">Private</Link>
      {isAuthenticated ? (
        <NormalButton onClick={() => logoutUser(() => history.push('/'))}>
          Logout
        </NormalButton>
      ) : (
        <>
          <Link to="/signup">Signup</Link>
          <Link to="/login">Login</Link>
        </>
      )}
    </>
  );
};

const mapStateToProps = state => ({
  isAuthenticated: selectIsAuthenticated(state),
});

export default withRouter(
  connect(
    mapStateToProps,
    { logoutUser }
  )(Header)
);
