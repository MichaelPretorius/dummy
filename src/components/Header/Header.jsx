import React, { useState } from 'react';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';

import { selectIsAuthenticated } from '../../redux/auth/authSelectors';
import { logoutUser } from '../../redux/auth/authActions';
import burger from '../../assets/navburger.png';
import {
  Button,
  NavContainer,
  Wrapper,
  Burger,
  NavBrand,
  Navbar,
  Navs,
  Nav,
} from '../../styles';

export const Header = ({ isAuthenticated, logoutUser, history }) => {
  const [showNav, setShowNav] = useState(false);

  const toggleNav = () => {
    setShowNav(!showNav);
  };

  return (
    <NavContainer row secondary>
      <Wrapper row>
        <Burger onClick={toggleNav} src={burger} />
        <NavBrand to="/">Home</NavBrand>
      </Wrapper>
      <Navbar showNav={showNav} width="200px">
        <Navs>
          <Nav to="/styles">Styles</Nav>
          <Nav to="/private">Private</Nav>
          {isAuthenticated ? (
            <Button onClick={() => logoutUser(() => history.push('/'))}>
              Logout
            </Button>
          ) : (
            <>
              <Nav to="/signup">Signup</Nav>
              <Nav to="/login">Login</Nav>
            </>
          )}
        </Navs>
      </Navbar>
    </NavContainer>
  );
};

const mapStateToProps = state => ({
  isAuthenticated: selectIsAuthenticated(state),
});

export default withRouter(
  connect(
    mapStateToProps,
    { logoutUser }
  )(Header)
);
