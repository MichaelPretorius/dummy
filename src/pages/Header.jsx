import React from 'react';
import { Link, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';

import { selectCurrentUser } from '../redux/auth/authSelectors';
import { NormalButton } from '../components/CustomButton';
import { logoutUser } from '../redux/auth/authActions';

const Header = ({ user, logoutUser, history }) => {
  return (
    <>
      <Link to="/">Home</Link>
      <Link to="/private">Private</Link>
      {user ? (
        <NormalButton onClick={() => logoutUser(() => history.push('/'))}>
          Logout
        </NormalButton>
      ) : (
        <>
          <Link to="/signup">Signup</Link>
          <Link to="/login">Login</Link>
        </>
      )}
    </>
  );
};

const mapStateToProps = state => ({
  user: selectCurrentUser(state),
});

export default withRouter(
  connect(
    mapStateToProps,
    { logoutUser }
  )(Header)
);
