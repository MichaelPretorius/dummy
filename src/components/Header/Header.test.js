import React from 'react';
import { shallow } from 'enzyme';

import { Header } from './Header';

describe('Header component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Header />);
  });

  test('should render Header component', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
