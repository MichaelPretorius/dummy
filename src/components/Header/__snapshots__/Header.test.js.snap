// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render Header component 1`] = `
"<NavContainer row={true} secondary={true}>
  <Wrapper row={true}>
    <Burger onClick={[Function: toggleNav]} src=\\"navburger.png\\" />
    <NavBrand to=\\"/\\">
      Home
    </NavBrand>
  </Wrapper>
  <Navbar showNav={false} width=\\"200px\\">
    <Navs>
      <Nav to=\\"/styles\\">
        Styles
      </Nav>
      <Nav to=\\"/private\\">
        Private
      </Nav>
      <Nav to=\\"/signup\\">
        Signup
      </Nav>
      <Nav to=\\"/login\\">
        Login
      </Nav>
    </Navs>
  </Navbar>
</NavContainer>"
`;
